src/api/transform/gemini-format.ts

```diff
@@ -16,6 +42,39 @@ export function convertAnthropicContentToGemini(content: string | Anthropic.Cont
 				}
 
 				return { inlineData: { data: block.source.data, mimeType: block.source.media_type } }
+			case "video": {
+				if (block.source.type !== "base64") {
+					throw new Error("Unsupported video source type. Only base64 encoded videos are supported.")
+				}
+
+				// Validate video MIME type
+				const supportedVideoTypes = ["video/mp4", "video/webm", "video/ogg", "video/quicktime"]
+				if (!supportedVideoTypes.includes(block.source.media_type)) {
+					throw new Error(
+						`Unsupported video format: ${block.source.media_type}. Supported formats: ${supportedVideoTypes.join(", ")}`,
+					)
+				}
+
+				// Check if video data exists
+				if (!block.source.data || block.source.data.trim() === "") {
```

Comment:
Based on the PR discussion, there's a < 20MB limitation for video files when using the HTTP API. Consider adding validation here to provide better error messages:
```suggestion
// Check if video data exists
if (!block.source.data || block.source.data.trim() === "") {
  throw new Error("Video data is empty or missing")
}

// Rough estimate: base64 is ~1.33x the original size
const estimatedSize = (block.source.data.length * 0.75) / (1024 * 1024); // in MB
if (estimatedSize > 20) {
  throw new Error(`Video size (~${estimatedSize.toFixed(1)}MB) exceeds the 20MB limit for direct API calls. Consider using the Files API for larger videos.`);
}
```